### ISSUE TITLE
indent.VariableDeclarators insufficient for different scenarios

### ISSUE THREAD

**Issue by ffxsam** [eslint/eslint#5492]
I ran into a problem just now, with this code:

```js
    const a = 'asdf',
          b = 'asdfjkhg';
    const arrayOfObjects = [
      {
        name: 'Bob',
        age: 32,
      },
      {
        name: 'Mary',
        age: 32,
      },
    ];
```

No matter how I set `VariableDeclarators`, I can't win. :) Either it complains about the alignment of variables `a` and `b`, or if I adjust for that, it complains about the curly braces.

IMO, and according to how I interpret the documentation, `VariableDeclarators` should only be concerned with indentation of variable names, *not* indentation of other syntax such as brackets and braces. In other words, the code above should be totally valid given this setting:

```js
    "indent": [2, 2, {
      "SwitchCase": 1,
      "VariableDeclarator": {"let": 2, "const": 3}
    }],
```

**Comment by eslintbot** [comment]
Hi @ffxsam, thanks for the issue. It looks like there's not enough information for us to know how to help you. If you're reporting a bug, please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
2. What you did (the source code and ESLint configuration)
3. The actual ESLint output complete with numbers
4. What you expected to happen instead

Requesting a new rule? Please see [Proposing a New Rule](http://eslint.org/docs/developer-guide/contributing/new-rules) for instructions.

If it's something else, please just provide as much additional information as possible. Thanks!

**Comment by ffxsam** [comment]
Version 2.2.0.

**Comment by gyandeeps** [comment]
Can you also provide
>The actual ESLint output complete with number

**Comment by ffxsam** [comment]
Sure. Here's the code. Line 25 is the first line:

```
    const varA = 'test',
          varB = 'test2';
    const others = [
      {
        thing: 'sky',
        color: 'blue',
      },
      {
        thing: 'dirt',
        color: 'brown',
      },
    ];
```

My .eslintrc is set to:

```
    "indent": [2, 2, {
      "SwitchCase": 1,
      "VariableDeclarator": {"let": 2, "const": 3}
    }],
```

I get the following errors:

```
  28:7  error  Expected indentation of 10 space characters but found 6  indent
  29:9  error  Expected indentation of 12 space characters but found 8  indent
  30:9  error  Expected indentation of 12 space characters but found 8  indent
  31:7  error  Expected indentation of 10 space characters but found 6  indent
  32:7  error  Expected indentation of 10 space characters but found 6  indent
  33:9  error  Expected indentation of 12 space characters but found 8  indent
  34:9  error  Expected indentation of 12 space characters but found 8  indent
  35:7  error  Expected indentation of 10 space characters but found 6  indent
```

If I modify the indent settings by commenting out VariableDeclarator:

```
    "indent": [2, 2, {
      "SwitchCase": 1//,
      //"VariableDeclarator": {"let": 2, "const": 3}
    }],
```

Then I get this:

```
  26:11  error  Expected indentation of 6 space characters but found 10  indent
```

There needs to be separate considerations for indenting/aligning variable names, and indenting braces/brackets.

**Comment by dgreensp** [comment]
I've been grappling with this problem over the past week, including with ESLint 2.3.0.  It "fixes" this code with excessive indentation:

```
     it('displays correct collapsed icon', function() {
       const instance = renderComponent(
-        {
-          collapsedIconName: 'collapsedIconName',
-          expandedIconName: 'expandedIconName'
-        },
+            {
+              collapsedIconName: 'collapsedIconName',
+              expandedIconName: 'expandedIconName'
+            },
         'FOOBAR');
```

**Comment by ffxsam** [comment]
Yep. Exactly why I think that `VariableDeclarator` needs to be modified so that it only cares about indentation of variable *names*:

```
const someLongVar = 3,
      anotherVar = 8;
```

But not braces, brackets, parens, and other non-alpha syntax:

```
  const someVar = [
    () => { stuff(); },
    () => { moreStuff(); },
  ];
```

**Comment by dgreensp** [comment]
I don't think that's exactly right; you still want:

```
const foo = 1,
      bar = [
        2
      ];
```

The offending logic seems to be: https://github.com/eslint/eslint/blob/bb8f47018681e7afa823165fdaec490e57955d53/lib/rules/indent.js#L448

I'm trying to figure out what this line is _supposed_ to do.

**Comment by dgreensp** [comment]
Ok, I have a PR ready for review: https://github.com/eslint/eslint/pull/5509
